### ISSUE TITLE
Use flexbox for download shortcode

### ISSUE THREAD

**Issue by SDavisMedia** [easydigitaldownloads/vendd#238]
I'm going to try. Wish me luck.

**Comment by SDavisMedia** [comment]
Not gonna happen the way I want with the current HTML structure. I've got some ideas, but I'll implement them at a later date.

**Comment by timeassistant** [comment]
I have built a product grid for a store page template that uses flex box, I'm fairly sure the page template would be pretty much the same as the downloads shortcode template. ITs very handy, it keeps the grid items at equal height and is natively responsive, all you may need is a media query for like 640px to swap it into a single column for mobiles. 

I modified mine from an EDD sample, either Vendd or the other one (cant quite remember) 

I would be happy to offer help as I am stuck with something else - could possibly trade help

**Comment by SDavisMedia** [comment]
Hi @timeassistant. I appreciate it, but there's no need to barter. If you are having issues with Vendd, feel free to open a support ticket. https://easydigitaldownloads.com/support/ If it's custom, we may not be able to help. But it definitely won't hurt to ask.

As for this issue, building a product grid that uses flexbox isn't the challenge. The actual goal is to make the [downloads] shortcode, which is standard for our users, as flexible as any custom layout would be. I'm moving away from custom product grids in my official EDD themes. They should all make use of our shortcode.

**Comment by timeassistant** [comment]
sorry wasn't meaning to give the hard sell there ! 

It's not vendd as such, I have modified this template : **http://docs.easydigitaldownloads.com/article/276-sample-product-grid-for-your-theme**

I was just wanting to add a download category filter to it and a tags filter to display downloads from specific categories, I know this can be done with the downloads shortcode but I am struggling to make it work via the template I think I could be over thinking it though!

as you suggested though I'll drop in a ticket see if anyone can help. 


Aside from that, the flex grid I've built uses most of the same standard coding, I moved a couple elements about and added some extra classes, then obviously stripped out things like display properties etc to swap in flex box properties. 

It works very well if I say so myself! in terms of flexibility, you could create .flex1 .flex2 .flex3 type classes that are added via PHP (via EDD interface or theme interface) which could allow users to set various options, such as how many columns etc. Or you could use an EDD or theme interface to program the flex properties direct using human readable wording which is then substituted for flex css properties when rendered etc. 

granted I'm not sure how you would do that for the downloads shortcode (I tried creating a template for it once and failed miserably!) which is why I'm using a page template now ;)

**Comment by SDavisMedia** [comment]
@timeassistant haha yea it's tough to do with the [downloads] shortcode. And that's exactly why this issue exists... specifically for the shortcode. If it was just for a page template, I would have done it in one sitting. Thanks for your input!

**Comment by timeassistant** [comment]
@SDavisMedia Yeah makes sense !  If there is ever a full template released for the download shortcode please update this thread ! I'd love to get my hands on one and edit it :)

**Comment by SDavisMedia** [comment]
Will end up going with CSS Grid instead. Changing the title.
